### ISSUE TITLE
Support nexus sync or management

### ISSUE THREAD

**Issue by nvandoorn** [SuperGoodSoft/solidus_taxjar#42]
This extension should provide a nexus management interface where users can add a new Taxjar nexus or sync their existing nexus list from Taxjar.

See more here: https://developers.taxjar.com/integrations/sales-tax-calculations/#section-nexus-addresses

**Acceptance criteria**

* [ ]  Sync all of the nexus locations currently associated with a Taxjar account to Solidus app if none exist
* [ ]  Display all of the nexus locations currently associated with a Taxjar account (This should be displayed on the [Taxjar settings page](https://github.com/SuperGoodSoft/solidus_taxjar/blob/master/app/views/spree/admin/taxjar_settings/show.html.erb))
* [ ]  Allow users to request a nexus sync on demand (The integration docs suggest a sync button)
* [ ]  Provide a way for users to add a new Taxjar nexus

**Comment by nvandoorn** [comment]
Work has started on this here: https://github.com/nvandoorn/solidus_taxjar/tree/add-taxjar-nexus-data-model

**Comment by Noah-Silvera** [comment]
It's possible for solidus to support multiple stores, and each store could have a different nexus regions.

Supporting multiple stores with different nexus regions for stores should be pretty straightforward, as all we would need to do is associate nexus regions with stores, and scope the places nexus regions are used accordingly.

This has significant business value as multi-store sites are not uncommon.

**Comment by Noah-Silvera** [comment]
@senemsoy this is a chunky one that could be good and spicy! It probably need to be broken down into smaller pieces before tackling.

**Comment by Noah-Silvera** [comment]
@senemsoy I made a new board for this work if you wanted to split it up and groom it before starting!

**Comment by forkata** [comment]
> Supporting multiple stores with different nexus regions for stores should be pretty straightforward, as all we would need to do is associate nexus regions with stores, and scope the places nexus regions are used accordingly.

Hey @Noah-Silvera, I have a quick question on this. Can this be managed within a single account on TaxJar, i.e. does TaxJar have a way for us to identify which store has which nexuses on their side or would this be entirely managed in Solidus and we would have to override the `nexus_addresses` list when we get taxes for an order based on the nexuses of the given store?

Either way this seems like a pretty advanced feature and I'd be curious if it's needed for certification 🤔

**Comment by Noah-Silvera** [comment]
Oof, excellent points Chris! A taxjar account has one set of nexuses associated with it (Nexus's are associated with a state), and to support multiple stores with multiple nexus's we'd either have to support multiple taxjar accounts, or do our own custom filtering per store in solidus as you suggested.... woof.

I feel it's safe to say that's out of scope for certification, and we can worry about associated nexus addresses with stores later if needed.

**Comment by benjaminwil** [comment]
The TaxJar developer documentation provides some information and a screenshot of the Magento implementation of the Nexus list functionality.

**Comment by benjaminwil** [comment]
This issue has been broken down and turned into two new issues:

- #156 
- #157
